### PR TITLE
linkerd2-proxy: add a new fuzzer and fix coverage build

### PR DESCRIPTION
**Pull request by DavidKorczynski**
Generalises build script to arbitrary amount of fuzzers as a reaction to new fuzzers usptream.

### CODE REVIEW

File: projects/linkerd2-proxy/build.sh [hunk 1]
```diff
@@ -14,27 +14,20 @@
...
 ################################################################################
-TARGET_PATH="./fuzz/target/x86_64-unknown-linux-gnu/release"
-BASE="$SRC/linkerd2-proxy/linkerd"
-BUILD_FUZZER="cargo +nightly fuzz build --features fuzzing"
 
-cd ${BASE}/addr/
-${BUILD_FUZZER}
-cp ${TARGET_PATH}/fuzz_target_1 $OUT/fuzz_addr
+BASE=$PWD
+FUZZERS=$(find . -name "fuzz")
```
Comment (asraa) [inline projects/linkerd2-proxy/build.sh:19]: do you want only directories?

Comment (DavidKorczynski) [reply projects/linkerd2-proxy/build.sh:19]: Yeah. All of the fuzzers are placed in "fuzz" directories across the repo and we need to find those in order to build the fuzzers.

Comment (DavidKorczynski) [reply projects/linkerd2-proxy/build.sh]: I had to add some more conditions for some tagrets, so decided to abandon generalising it in favour of readability. However, it should be ready to go now!

**Review by AdamKorcz — APPROVED** [review]
LGTM!

**Pull request merged**